#! /bin/sh /usr/share/dpatch/dpatch-run
## 01_fix_fontcache_nullderef.dpatch by A Mennucc <mennucc1@debian.org>.
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fixes NULL dereference in gdkglfont-x11.c.

@DPATCH@

--- gtkglext-1.0.6/gdk/x11/gdkglfont-x11.c	2004-02-20 10:56:23.000000000 +0100
+++ gtkglext-1.0.6.mine/gdk/x11/gdkglfont-x11.c	2006-05-12 11:40:11.416117472 +0200
@@ -111,6 +111,12 @@
 
   fs = pango_x_font_cache_load (font_cache, xlfd);
 
+  if( fs == NULL) {
+    g_warning ("cannot get X font cache");
+    font = NULL;
+    goto FAIL;
+  }
+
   glXUseXFont (fs->fid, first, count, list_base);
 
   pango_x_font_cache_unload (font_cache, fs);
